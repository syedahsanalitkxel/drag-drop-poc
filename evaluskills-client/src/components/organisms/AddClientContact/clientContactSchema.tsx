import * as Yup from 'yup';
import { alphabetsValidation, emailValidation, stringValidation } from '../../../utils/validations';

const clientContactSchema = Yup.object().shape({
  email: emailValidation(true),
  firstName: alphabetsValidation(2, 250, true),
  lastName: alphabetsValidation(2, 250, true),
  title: stringValidation(1, 250, true),
});

export default clientContactSchema;
