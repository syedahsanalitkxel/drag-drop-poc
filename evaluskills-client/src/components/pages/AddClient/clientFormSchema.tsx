import * as Yup from 'yup';
import { stringValidation } from '../../../utils/validations';

const clientFormSchema = Yup.object().shape({
  address: stringValidation(2, 250, true),
  billing: stringValidation(2, 250, true),
  city: stringValidation(1, 250, true),
  clientInformation: stringValidation(2, 250, true),
  clientName: stringValidation(1, 250, true),
  clientType: stringValidation(1, 250, true),
  phone: stringValidation(2, 50, true),
  school: stringValidation(2, 250, true),
  state: stringValidation(2, 250, true),
  userEmail: stringValidation(2, 50, true),
  userFirstName: stringValidation(2, 50, true),
  userLastName: stringValidation(2, 50, true),
  zip: stringValidation(2, 50, true),
});

export default clientFormSchema;
