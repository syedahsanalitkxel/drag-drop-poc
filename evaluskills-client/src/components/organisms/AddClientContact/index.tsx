import { Field, Formik } from 'formik';
import React from 'react';
import { Button, Modal, ModalBody, ModalHeader } from 'reactstrap';
import clientInterface, { ContactInterface } from '../../../interfaces/Client';
import { styles } from '../../pages/AddUser/style';
import clientContactSchema from './clientContactSchema';
import FormikBag from '../../../interfaces/FormikBag';
import PageBody from '../../atoms/PageBody';
import FormElement, { FormElementTypes } from '../../molecules/FormElement';
import styled from 'styled-components';

interface Props {
  index?: number;
  fprops: any;
  children?: React.ReactNode;
  visible?: boolean;
  name?: string;
  toggle?: () => void;
  formStateUpdate: (values: any) => void;
  formValues?: any;
}

const StyledButton = styled(Button)`
  margin-left: 5px;
  margin-right: 5px;
`;

export const AddClientContact: React.FunctionComponent<Props> = ({
  visible,
  toggle,
  index,
  fprops,
  formValues,
  formStateUpdate,
  name,
}) => {
  function submitHandler(values: any) {
    if (fprops.clientContacts && toggle && name === 'Add' && values) {
      fprops.clientContacts.push(values);
      formStateUpdate(fprops);
      toggle();
    } else if (fprops.clientContacts && toggle && name === 'Edit') {
      const contactIndex = fprops.clientContacts.findIndex(
        (contact: any) => contact.id === values.id
      );
      fprops.clientContacts[contactIndex] = values;
      formStateUpdate(fprops);
      toggle();
    }
  }

  const renderForm = (formikprops: FormikBag) => {
    return (
      <form onSubmit={formikprops.handleSubmit.bind(formikprops)} className={'form'}>
        <div className="row">
          <div className="col-md-6">
            <FormElement
              label="First Name"
              name="firstName"
              placeholder="Add First Name"
              formikprops={formikprops}
              inline={true}
            />
          </div>
          <div className="col-md-6">
            <FormElement
              label="Last Name"
              name="lastName"
              placeholder="Add Last Name"
              formikprops={formikprops}
              inline={true}
            />
          </div>
        </div>

        <div className="row">
          <div className="col-md-6">
            <FormElement
              label="Email"
              name="email"
              placeholder="Add Email"
              formikprops={formikprops}
              inline={true}
            />
          </div>
          <div className="col-md-6">
            <FormElement
              label="Phone"
              name="phone"
              placeholder="Add Phone"
              formikprops={formikprops}
              noValidate={true}
              inline={true}
            />
          </div>
        </div>

        <div className="row">
          <div className="col-md-6">
            <FormElement
              label="Role"
              name="title"
              formikprops={formikprops}
              inline={true}
              last={true}
            />
          </div>
        </div>

        <PageBody>
          <div className="row m-b-2">
            <StyledButton type="button" size="lg">
              Cancel
            </StyledButton>
            <StyledButton type="submit" color="primary" size="lg">
              Save
            </StyledButton>
          </div>
        </PageBody>
      </form>
    );
  };
  return (
    <Modal isOpen={visible} toggle={toggle} style={styles.modal_width}>
      <ModalHeader toggle={toggle}>{name} Contact</ModalHeader>
      <ModalBody>
        <Formik
          initialValues={formValues}
          validationSchema={clientContactSchema}
          onSubmit={submitHandler}
        >
          {formikprops => renderForm(formikprops)}
        </Formik>
      </ModalBody>
    </Modal>
  );
};

AddClientContact.defaultProps = {
  visible: false,
};

export default AddClientContact;
