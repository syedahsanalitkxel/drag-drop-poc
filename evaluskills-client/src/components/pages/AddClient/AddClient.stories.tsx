import React, { useState } from 'react';

import { storiesOf } from '@storybook/react';
import AddClientContact from '../../organisms/AddClientContact/index';
import AddClient from './index';
import EditClientContact from '../../organisms/AddClientContact/index';
import { Simulate } from 'react-dom/test-utils';
import submit = Simulate.submit;

const formValues = {
  address: 'Cantt. MughalPura Lahore',
  billing: '',
  city: 'Lahore',
  clientInformation: '',
  clientName: 'TkXel',
  clientType: '',
  contact: [
    {
      email: 'maria@evaluskills.com',
      firstName: 'Maria',
      id: '1',
      lastName: ' Garcia',
      phone: '+1 818-452-1505',
      role: '',
    },
  ],
  id: '2',
  noOfAssessments: '25',
  noOfEvaluators: '28',
  noOfParticipants: '30',
  phone: '+1 818-452-1505',
  plan: 'Plan 01',
  school: 'P.R Boys high school',
  state: 'punjab',
  status: 'Active',
  userEmail: 'ali@tkxel.com',
  userFirstName: 'ali',
  userLastName: 'raza',
  zip: '54000',
};

interface ModalProps {
  index?: number;
  fprops: any;
  children?: React.ReactNode;
  visible?: boolean;
  name?: string;
  toggle?: () => void;
  formValues?: any;
}
const modalValues = {
  email: 'maria@evaluskills.com',
  firstName: 'maria',
  id: '1',
  lastName: 'gracia',
  role: 'admin',
};

const [addClientContactModalVisible, setAddClientContactModalVisible] = useState(false);
const [editClientContactModalVisible, setEditClientContactModalVisible] = useState(false);

const toggleAddClientContactModal = () => {
  setAddClientContactModalVisible(!addClientContactModalVisible);
};

const toggleEditClientContactModal = () => {
  setEditClientContactModalVisible(!editClientContactModalVisible);
};

function submitForm(values: any) {
  delete values.clientContacts;
  values.clientContacts = values.contact;
  console.log(values);
}

const addContactProps: ModalProps = {
  formValues: modalValues,
  fprops: formValues,
  name: 'Add',
  toggle: toggleAddClientContactModal,
  visible: true,
};

const editContactProps: ModalProps = {
  formValues: modalValues,
  fprops: formValues,
  name: 'Edit',
  toggle: toggleEditClientContactModal,
  visible: true,
};

storiesOf('AddClient', module)
  .add('AddClient', () => <AddClient changeListener={submitForm} />)
  .add('AddClientContact renders with props', () => <AddClientContact {...addContactProps} />)
  .add('EditClientContact renders with props', () => <EditClientContact {...editContactProps} />);
