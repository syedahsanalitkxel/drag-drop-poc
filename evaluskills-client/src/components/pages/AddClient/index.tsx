import React, { Fragment, useEffect, useState } from 'react';

import { Field, Formik, FormikActions, FormikValues } from 'formik';
import { Button, Form, FormFeedback, FormGroup, Input, Label } from 'reactstrap';
import styled from 'styled-components';

import AddClientInterface from '../../../interfaces/AddClient';
import FormikBag from '../../../interfaces/FormikBag';
import PageBody from '../../atoms/PageBody';
import FormElement, { FormElementTypes } from '../../molecules/FormElement';
import ClientContacts from '../../organisms/ClientContacts/ClientContacts';
import DashboardTemplate from '../../templates/DashboardTemplate';
import clientFormSchema from './clientFormSchema';

interface Props {
  changeListener?: (formValues: AddClientInterface) => void;
}

const initialState: AddClientInterface = {
  address: '',
  billing: 'plan 2',
  city: '',
  clientInformation: '',
  clientName: 'Maria Gracia',
  clientType: 'Type 1',
  contact: [
    {
      email: 'Ali@tkxel.com',
      firstName: 'Ali',
      lastName: 'Raza',
      phone: '+923334567891',
      role: '',
    },
  ],
  phone: '+888 667 999 ',
  school: '',
  state: '',
  userEmail: 'rizwan@tkxel.com',
  userFirstName: 'rizwan',
  userLastName: 'shah',
  zip: '',
};

const StyledButton = styled(Button)`
  margin-left: 5px;
  margin-right: 5px;
`;

export const AddClient: React.FunctionComponent<Props> = ({ changeListener }) => {
  const [formState, setFormState] = useState(initialState);

  useEffect(() => {
    if (changeListener) {
      changeListener(formState);
    }
  });

  function submitForm(
    values: AddClientInterface,
    formikActions: FormikActions<AddClientInterface>
  ) {
    setFormState({ ...formState, ...values });
  }

  const onClickAddContact = (event: React.MouseEvent) => {
    event.preventDefault();

    const { contact } = formState;
    const contactObj: any = {};
    if (contact) {
      contact.push(contactObj);
    }
    setFormState({ ...formState, contact });
  };

  const renderForm = (formikprops: FormikBag) => {
    const renderContactList = (contact: any, index: number) => (
      <Fragment key={index}>
        <ClientContacts formikprops={formikprops} />
      </Fragment>
    );

    return (
      <Form onSubmit={formikprops.handleSubmit} className="form">
        <PageBody card={true} className="m-t-15">
          <FormElement
            label="Client Name"
            name="clientName"
            placeholder="Add Client Name"
            formikprops={formikprops}
          />

          <FormElement
            label="Logo"
            name="logo"
            formikprops={formikprops}
            noValidate={true}
            type={FormElementTypes.IMAGE_UPLOAD}
          />

          <FormElement
            label="Address"
            name="address"
            placeholder="Add Address"
            formikprops={formikprops}
          />

          <FormElement label="City" name="city" placeholder="Add City" formikprops={formikprops} />

          <FormElement
            label="State"
            name="state"
            placeholder="Add State"
            formikprops={formikprops}
          />

          <FormElement label="Zip" name="zip" placeholder="Add Zip" formikprops={formikprops} />

          <FormElement
            label="School/Subsidiary"
            name="school"
            placeholder="Add School"
            formikprops={formikprops}
          />

          <FormElement
            label="Client Information"
            name="clientInformation"
            placeholder="Add Client Information"
            formikprops={formikprops}
          />

          <FormElement
            label="Billing"
            name="billing"
            formikprops={formikprops}
            type={FormElementTypes.SELECT}
          >
            <option value="billing-1">Biling 1</option>
            <option value="billing-2">Biling 2</option>
          </FormElement>

          <FormElement
            label="Client Type"
            name="clientType"
            formikprops={formikprops}
            type={FormElementTypes.SELECT}
          >
            <option value="selected">Select Type</option>
            <option value="co-oprate">Co-oprate</option>
            <option value="Educational Institute">Educational Institute</option>
          </FormElement>
        </PageBody>

        <div className="form-header row">
          <div className="col-sm-6">
            <h2>Contact</h2>
          </div>
          <div className="col-sm-6">
            <Button
              className="mt-3 float-right"
              color="primary"
              size="lg"
              onClick={onClickAddContact}
            >
              Add Contact
            </Button>
          </div>
        </div>

        <div>{formState.contact && formState.contact.map(renderContactList)}</div>

        <div className="form-header row">
          <div className="col-sm-6">
            <h2>User Information</h2>
          </div>
        </div>

        <PageBody card={true}>
          <FormElement
            label="First Name"
            name="userFirstName"
            placeholder="Add First Name"
            formikprops={formikprops}
          />
          <FormElement
            label="Last Name"
            name="userLastName"
            placeholder="Add Last Name"
            formikprops={formikprops}
          />
          <FormElement
            label="Email"
            name="userEmail"
            placeholder="Add Email"
            formikprops={formikprops}
            last={true}
          />
        </PageBody>

        <PageBody>
          <div className="row m-b-25">
            <StyledButton type="button" size="lg">
              Cancel
            </StyledButton>
            <StyledButton type="submit" color="primary" size="lg">
              Save
            </StyledButton>
            <StyledButton type="button" color="primary" size="lg">
              Save &amp; Add More
            </StyledButton>
          </div>
        </PageBody>
      </Form>
    );
  };

  return (
    <DashboardTemplate>
      <Formik initialValues={formState} validationSchema={clientFormSchema} onSubmit={submitForm}>
        {(formikprops: FormikBag) => renderForm(formikprops)}
      </Formik>
    </DashboardTemplate>
  );
};

export default AddClient;
