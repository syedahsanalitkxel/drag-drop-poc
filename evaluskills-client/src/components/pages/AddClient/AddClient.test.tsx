import { configure, mount, render, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import AddClientContact from '../../organisms/AddClientContact/index';
import AddClient from './index';
import EditClientContact from '../../organisms/AddClientContact/index';

configure({ adapter: new Adapter() });

const formValues = {
  address: 'Cantt. MughalPura Lahore',
  billing: '',
  city: 'Lahore',
  clientInformation: '',
  clientName: 'TkXel',
  clientType: '',
  contact: [
    {
      email: 'maria@evaluskills.com',
      firstName: 'Maria',
      id: '1',
      lastName: ' Garcia',
      phone: '+1 818-452-1505',
      role: '',
    },
  ],
  id: '2',
  noOfAssessments: '25',
  noOfEvaluators: '28',
  noOfParticipants: '30',
  phone: '+1 818-452-1505',
  plan: 'Plan 01',
  school: 'P.R Boys high school',
  state: 'punjab',
  status: 'Active',
  userEmail: 'ali@tkxel.com',
  userFirstName: 'ali',
  userLastName: 'raza',
  zip: '54000',
};

describe('MyComponent', () => {
  const action = 'edit';
  const props = { defaultValue: { formValues }, action: { action } };
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<AddClient />);
    expect(component).toMatchSnapshot();
  });
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Router>
        <AddClient />
      </Router>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it('to be defined of form control `.form`s', () => {
    const wrapper = shallow(<AddClient />);
    expect(wrapper.find('.form')).toBeDefined();
  });
  it('to be defined of AddClientContacts', () => {
    const wrapper = shallow(<AddClient />);
    expect(wrapper.find('AddClientContacts')).toBeDefined();
  });
  it('to be defined of EditClientContacts', () => {
    const wrapper = shallow(<AddClient />);
    expect(wrapper.find('EditClientContacts')).toBeDefined();
  });
  it('to be defined of Formik Form', () => {
    const wrapper = shallow(<AddClient />);
    expect(wrapper.find('Formik')).toBeDefined();
  });
  it('16 Input fields in `.form`', () => {
    const wrapper = mount(
      <Router>
        <AddClient />
      </Router>
    );
    expect(wrapper.find('Input')).toHaveLength(16);
  });
  it('17 Form Feedback in `.form`', () => {
    const wrapper = mount(
      <Router>
        <AddClient />
      </Router>
    );
    expect(wrapper.find('FormFeedback')).toHaveLength(17);
  });
  it('48 form control `.form-control`s', () => {
    const wrapper = mount(
      <Router>
        <AddClient />
      </Router>
    );
    expect(wrapper.find('.form-control')).toHaveLength(48);
    expect(wrapper.find('.form-control')).toBeDefined();
  });
  it('First Name label exist ', () => {
    const wrapper = render(
      <Router>
        <AddClient />
      </Router>
    );
    expect(wrapper.text()).toContain('First Name');
  });
  it('Last Name label exist ', () => {
    const wrapper = render(
      <Router>
        <AddClient />
      </Router>
    );
    expect(wrapper.text()).toContain('Last Name');
  });
  it('Email label exist ', () => {
    const wrapper = render(
      <Router>
        <AddClient />
      </Router>
    );
    expect(wrapper.text()).toContain('Email');
  });
  it('Role label exist ', () => {
    const wrapper = render(
      <Router>
        <AddClient />
      </Router>
    );
    expect(wrapper.text()).toContain('Role');
  });
  it('Address label exist ', () => {
    const wrapper = render(
      <Router>
        <AddClient />
      </Router>
    );
    expect(wrapper.text()).toContain('Address');
  });
  it('City label exist ', () => {
    const wrapper = render(
      <Router>
        <AddClient />
      </Router>
    );
    expect(wrapper.text()).toContain('City');
  });
  it('State label exist ', () => {
    const wrapper = render(
      <Router>
        <AddClient />
      </Router>
    );
    expect(wrapper.text()).toContain('State');
  });
  it('Zip label exist ', () => {
    const wrapper = render(
      <Router>
        <AddClient />
      </Router>
    );
    expect(wrapper.text()).toContain('Zip');
  });
  it('Billing label exist ', () => {
    const wrapper = render(
      <Router>
        <AddClient />
      </Router>
    );
    expect(wrapper.text()).toContain('Billing');
  });
  it('School/Subsidiary label exist ', () => {
    const wrapper = render(
      <Router>
        <AddClient />
      </Router>
    );
    expect(wrapper.text()).toContain('School/Subsidiary');
  });
  it('Client Name label exist ', () => {
    const wrapper = render(
      <Router>
        <AddClient />
      </Router>
    );
    expect(wrapper.text()).toContain('Client Name');
  });
  it('Client Type ', () => {
    const wrapper = render(
      <Router>
        <AddClient />
      </Router>
    );
    expect(wrapper.text()).toContain('Client Type');
  });
  it('Contact label exist ', () => {
    const wrapper = render(
      <Router>
        <AddClient />
      </Router>
    );
    expect(wrapper.text()).toContain('Contact');
  });
  it('User Information label exist ', () => {
    const wrapper = render(
      <Router>
        <AddClient />
      </Router>
    );
    expect(wrapper.text()).toContain('User Information');
  });

  // describe('when typing into the First Name input', () => {
  //   const person = 'Uncle';
  //   const wrapper = render(
  //     <Router>
  //       <AddClient />
  //     </Router>
  //   );
  //   console.log(wrapper)
  //   beforeEach( () => {
  //     wrapper.find('.userFirstName-input').simulate('change', { target : { value: person } });
  //   })
  //
  //   it ('update the person in `state`', () => {
  //     expect(wrapper.state().userFirstName).toEqual(person)
  //   })
  // })
  //

  it('submit form ', () => {
    const wrapper = mount(
      <Router>
        <AddClient />
      </Router>
    );
    wrapper.find('form').simulate('submit');
  });
  it('submit form ', () => {
    const wrapper = mount(
      <Router>
        <AddClient />
      </Router>
    );
    wrapper.find('form').simulate('submit', {
      preventDefault: (clientProps: any) => {
        console.log(clientProps);
      },
    });
  });
  //
  describe('Component functionality', () => {
    it('should not submit if required fields are empty', () => {
      const badFormValues = {
        address: ' ',
        billing: '',
        city: '',
        clientInformation: '',
        clientName: '',
        clientType: '',
        contact: [
          {
            email: '',
            firstName: '',
            id: '',
            lastName: ' ',
            phone: '',
            role: '',
          },
        ],
        id: '',
        noOfAssessments: '',
        noOfEvaluators: '',
        noOfParticipants: '',
        phone: '',
        plan: '',
        school: '',
        state: '',
        status: '',
        userEmail: '',
        userFirstName: '',
        userLastName: '',
        zip: '',
      };
      const resetForm = jest.fn();
      const componentRender = mount(
        <Router>
          <AddClient {...badFormValues} />
        </Router>
      );
      componentRender
        .find('Formik')
        .instance()
        .props.onSubmit(badFormValues, { resetForm });
      // console.log(componentRender.find('FormFeedback').debug());
      // console.log(componentRender.update().find('.validation-error'));
      // console.log(componentRender.find('Formik').instance());
      // expect(componentRender.find('.validation-error').text()).toEqual('Company Name is required');
    });

    describe('AddClientContact', () => {
      interface ModalProps {
        index?: number;
        fprops: any;
        children?: React.ReactNode;
        visible?: boolean;
        name?: string;
        toggle?: () => void;
        formValues?: any;
      }
      const toggleModal = jest.fn();
      const modalValues = {
        email: 'maria@evaluskills.com',
        firstName: 'maria',
        id: '1',
        lastName: 'gracia',
        role: 'admin',
      };
      const addContactProps: ModalProps = {
        formValues: modalValues,
        fprops: formValues,
        name: 'Add',
        toggle: toggleModal,
        visible: true,
      };

      it('to be Contain of AddClientContacts render without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
          <Router>
            <AddClientContact {...addContactProps} />
          </Router>,
          div
        );
        ReactDOM.unmountComponentAtNode(div);
      });
      it('to be defined of form control `.form`s', () => {
        const wrapper = shallow(<AddClientContact {...addContactProps} />);
        expect(wrapper.find('.form')).toBeDefined();
      });
      it('5 Input fields in `.form`', () => {
        const wrapper = mount(
          <Router>
            <AddClientContact {...addContactProps} />
          </Router>
        );
        expect(wrapper.find('Input')).toHaveLength(5);
      });
      it('5 Form Feedback in `.form`', () => {
        const wrapper = mount(
          <Router>
            <AddClientContact {...addContactProps} />
          </Router>
        );
        expect(wrapper.find('FormFeedback')).toHaveLength(5);
      });
      it('15 form control `.form-control`s', () => {
        const wrapper = mount(
          <Router>
            <AddClientContact {...addContactProps} />
          </Router>
        );
        expect(wrapper.find('.form-control')).toHaveLength(15);
        expect(wrapper.find('.form-control')).toBeDefined();
      });
    });

    describe('EditClientContact', () => {
      interface ModalProps {
        index?: number;
        fprops: any;
        children?: React.ReactNode;
        visible?: boolean;
        name?: string;
        toggle?: () => void;
        formValues?: any;
      }
      const toggleModal = jest.fn();
      const modalValues = {
        email: 'maria@evaluskills.com',
        firstName: 'maria',
        id: '1',
        lastName: 'gracia',
        role: 'admin',
      };
      const editContactProps: ModalProps = {
        formValues: modalValues,
        fprops: formValues,
        name: 'Edit',
        toggle: toggleModal,
        visible: true,
      };

      it('to be Contain of EditClientContact render without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
          <Router>
            <EditClientContact {...editContactProps} />
          </Router>,
          div
        );
        ReactDOM.unmountComponentAtNode(div);
      });
      it('to be defined of form control `.form`s', () => {
        const wrapper = shallow(<EditClientContact {...editContactProps} />);
        expect(wrapper.find('.form')).toBeDefined();
      });
      it('5 Input fields in `.form`', () => {
        const wrapper = mount(
          <Router>
            <EditClientContact {...editContactProps} />
          </Router>
        );
        expect(wrapper.find('Input')).toHaveLength(5);
      });
      it('5 Form Feedback in `.form`', () => {
        const wrapper = mount(
          <Router>
            <EditClientContact {...editContactProps} />
          </Router>
        );
        expect(wrapper.find('FormFeedback')).toHaveLength(5);
      });
      it('15 form control `.form-control`s', () => {
        const wrapper = mount(
          <Router>
            <EditClientContact {...editContactProps} />
          </Router>
        );
        expect(wrapper.find('.form-control')).toHaveLength(15);
        expect(wrapper.find('.form-control')).toBeDefined();
      });
    });
  });
});
