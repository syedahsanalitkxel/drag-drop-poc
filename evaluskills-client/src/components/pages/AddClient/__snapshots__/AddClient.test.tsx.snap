// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
<DashboardTemplate>
  <Formik
    enableReinitialize={false}
    initialValues={
      Object {
        "address": " ",
        "billing": " ",
        "city": " ",
        "clientInformation": " ",
        "clientName": " ",
        "clientType": " ",
        "contact": Array [
          Object {
            "email": " ",
            "firstName": " ",
            "id": " ",
            "lastName": " ",
            "phone": " ",
            "role": " ",
          },
        ],
        "id": " ",
        "noOfAssessments": " ",
        "noOfEvaluators": " ",
        "noOfParticipants": " ",
        "phone": " ",
        "plan": " ",
        "school": " ",
        "state": " ",
        "status": " ",
        "userEmail": " ",
        "userFirstName": " ",
        "userLastName": " ",
        "zip": " ",
      }
    }
    isInitialValid={false}
    onSubmit={[Function]}
    validateOnBlur={true}
    validateOnChange={true}
    validationSchema={
      ObjectSchema {
        "_blacklist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "_conditions": Array [],
        "_defaultDefault": [Function],
        "_deps": Array [],
        "_excludedEdges": Array [],
        "_exclusive": Object {},
        "_mutate": undefined,
        "_nodes": Array [
          "zip",
          "userLastName",
          "userFirstName",
          "userEmail",
          "state",
          "school",
          "phone",
          "clientType",
          "clientName",
          "clientInformation",
          "city",
          "billing",
          "address",
        ],
        "_options": Object {
          "abortEarly": true,
          "recursive": true,
        },
        "_type": "object",
        "_typeError": [Function],
        "_whitelist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "fields": Object {
          "address": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "min": true,
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "billing": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "min": true,
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "city": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "min": true,
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "clientInformation": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "min": true,
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "clientName": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "min": true,
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "clientType": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "min": true,
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "phone": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "min": true,
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "school": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "min": true,
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "state": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "min": true,
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "userEmail": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "min": true,
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "userFirstName": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "min": true,
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "userLastName": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "min": true,
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "zip": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "min": true,
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
      }
    }
  >
    <Component />
  </Formik>
</DashboardTemplate>
`;
